api/scpca_portal/models/contact.py

```diff
@@ -15,3 +21,44 @@ class Meta:
 
     def __str__(self) -> str:
         return f"{self.name} <{self.email}>"
+
+    @classmethod
+    def get_from_dict(cls, data: Dict):
+        contact = cls(
+            name=data.get("name"),
+            email=data.get("email"),
+            pi_name=data.get("pi_name"),
+        )
+
+        return contact
+
+    @staticmethod
+    def bulk_create_from_project_data(project_data, project):
```

Comment:
I was thinking about this a little more and wanted to suggest a new utility method.

```python
def get_delimited_values_as_dicts(data, *args, delimiter=common.CSV_MULTI_VALUE_DELIMITER):
    # Split the values we care about by their delimiter
    split_values = [data[key].split(delimiter) for key in args]
    # collect related values into their own list
    zipped_delimited_values = zip(*split_values, strict=True)
    # create a list of dictionaries that contain zipped key value pairs
    return [dict(zip(args, values)) for values in split_values]
```
The other alternative would be to return the `zipped_delimited_values` (converted to a list probably) and then still do another iteration on them and construct your own dictionaries.

I think this would clean up a lot of the details for each out this could be passed into the `get_from_dict` factory methods, but we would still want to clean up or validate the inputs. We would also want to catch the error from a length mismatch.

Reply:
I looked into this. Because of the varied nature of the data dictionary value transformations (some send parameters to `strip()`, others use `utils.boolean_from_string()`, another is a `project.pi_name` attribute), it was hard to take the extra step and to make the dictionary for `get_from_dict` truly generic, as proposed here. Check out the direction I took and let me know if you think there's a way we can organize better.

Reply:
If you like the utils function that I added, then I'll write tests for it and we can close this guy up.

Reply:
Yeah I think this is fine as an approach